containers-tests/tests/intmap-properties.hs

```diff
@@ -237,36 +246,55 @@ tests = [ testGroup "Test Case" [
 -- Operators
 
 test_index :: Assertion
-test_index = fromList [(5,'a'), (3,'b')] ! 5 @?= 'a'
+test_index = do
+    fromList [(5,'a'), (-3,'b')] ! 5 @?= 'a'
+
+    fromList [(5,'a'), (-3,'b')] ! (-3) @?= 'b'
 
 test_index_lookup :: Assertion
 test_index_lookup = do
     fromList [(5,'a'), (3,'b')] !? 1 @?= Nothing
     fromList [(5,'a'), (3,'b')] !? 5 @?= Just 'a'
 
+    fromList [(5,'a'), (-3,'b')] !? 1 @?= Nothing
+    fromList [(5,'a'), (-3,'b')] !? 5 @?= Just 'a'
+    fromList [(5,'a'), (-3,'b')] !? (-3) @?= Just 'b'
+
 ----------------------------------------------------------------
 -- Query
 
 test_size :: Assertion
 test_size = do
-    null (empty)           @?= True
-    null (singleton 1 'a') @?= False
+    null (empty)              @?= True
+    null (singleton 1 'a')    @?= False
+
+    null (singleton (-1) 'a') @?= False
 
 test_size2 :: Assertion
 test_size2 = do
     size empty                                   @?= 0
     size (singleton 1 'a')                       @?= 1
     size (fromList([(1,'a'), (2,'c'), (3,'b')])) @?= 3
 
+    size (fromList [(-2, '?'),(5,'a'), (3,'b')]) @?= 3
+
 test_member :: Assertion
 test_member = do
-    member 5 (fromList [(5,'a'), (3,'b')]) @?= True
-    member 1 (fromList [(5,'a'), (3,'b')]) @?= False
+    member 5    (fromList [(5,'a'), (3,'b')])  @?= True
+    member 1    (fromList [(5,'a'), (3,'b')])  @?= False
```

Comment:
Please restore the spacing so the diff will be more readable. I care *much more* about that than about things lining up.